src/inner.rs

```diff
@@ -2,10 +2,12 @@ use std::fmt;
 use std::hash::{BuildHasher, Hash};
 use std::mem::ManuallyDrop;
 
-#[cfg(feature = "indexed")]
-pub(crate) use indexmap::IndexMap as MapImpl;
-#[cfg(not(feature = "indexed"))]
-pub(crate) use std::collections::HashMap as MapImpl;
+#[cfg(all(feature = "indexed", not(feature = "amortize")))]
+pub(crate) use indexmap::{map::Entry, IndexMap as MapImpl};
+#[cfg(feature = "amortize")]
+pub(crate) use indexmap_amortized::{map::Entry, IndexMap as MapImpl};
```

Comment:
Seeing how concentrated this change is is crazy cool. Having this `MapImpl` idea seems super useful in this way. General question - did you plan that from the start? Or did that become clear to you the first time you needed to sub in another implementation? If that was something you planned from the start - was it just something clever you thought of or are there some general ideas stored anywhere that are useful for people to consider when building things? This seems like a cool thing to know about when building a lib.

I know this PR isn't a Q&A but I'm not sure what other medium to use to pick your brain 🤔 Can definitely take this offline somewhere if more convenient/keeps the PR discussion clean + focused

Reply:
Happy to do it here :)
No, I didn't think of this from the start — it just seemed like a handy way to avoid repeating the same config all over the place once I needed more than one impl!

Reply:
Coolio - good to know.